pages/index.tsx

```diff
@@ -1,18 +1,138 @@
-import React from 'react'
+import React, { useState, useEffect, useCallback } from 'react'
 import Page from '../client/components/page'
 
 // Polaris
-import { DisplayText, Layout } from '@shopify/polaris'
+import {
+	DisplayText,
+	Layout,
+	Button,
+	Checkbox,
+	Form,
+	FormLayout,
+	TextField
+} from '@shopify/polaris'
 
 // Events
 import Events from '../client/components/events'
-import { IndexProps } from '../shared/types'
-import { fetchRawEvents, enrichEvents } from '../client/events'
+import { IndexProps, RichEventType, RichEventsType } from '../shared/types'
+import { fetchRawEvents, enrichEvents, fetchRawEventDb } from '../client/events'
 import { IncomingMessage } from 'http'
 
+import Daet from 'daet'
+
+let hostname: string = ''
+let enableFauna: boolean = false
+
+function AddNewEventForum() {
+	const now = new Daet()
+	const [eventName, setEventName] = useState('')
+	const [eventDesc, setEventDesc] = useState('')
+	/* Use default so that use has something to work with and dont have to
+	   type this really long date string. */
+	const [startTime, setStartTime] = useState(now.toISOString())
+	const [endTime, setEndTime] = useState(now.toISOString())
+
+	const handleSubmit = useCallback(
+		_event => {
+			const req = new XMLHttpRequest()
```

Comment:
XMLHttpRequest is an old API, new API is https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API